### PR TITLE
P1754R1 Rename concepts to standard_case for C++20

### PR DESCRIPTION
**Pull request by burblebee**

### CODE REVIEW

File: source/utilities.tex [hunk 1]
```diff
@@ -8221,11 +8221,11 @@
 \begin{itemdecl}
 namespace ranges {
   template<@\placeholdernc{no-throw-forward-iterator}@ I, @\placeholdernc{no-throw-sentinel}@<I> S>
-      requires DefaultConstructible<iter_value_t<I>>
-    I uninitialized_default_construct(I first, S last);
+    requires default_constructible<iter_value_t<I>>
+      I uninitialized_default_construct(I first, S last);
```
[the remark this answers is not in the document]
Comment (CaseyCarter) [reply source/utilities.tex]: This is definitely the style we used for the majority of the declarations with *requires-clause*s introduced by P0896R4; apparently these were missed.

File: source/algorithms.tex [hunk 2]
```diff
@@ -1434,33 +1434,35 @@
     template<class I, class O>
     using replace_copy_result = copy_result<I, O>;
 
-    template<InputIterator I, Sentinel<I> S, class T1, class T2, OutputIterator<const T2&> O,
+    template<input_iterator I, sentinel_for<I> S,
+             class T1, class T2, output_iterator<const T2&> O,
              class Proj = identity>
```
Comment (CaseyCarter) [inline source/algorithms.tex]: Any reason not to two-line this *template-head* a la:
```c++
template<input_iterator I, sentinel_for<I> S, class T1, class T2, 
         output_iterator<const T2&> O, class Proj = identity>
```

File: source/algorithms.tex [hunk 3]
```diff
@@ -1650,20 +1652,20 @@
...
-             class Proj = identity, IndirectRelation<projected<I, Proj>> C = ranges::equal_to>
-      requires IndirectlyCopyable<I, O> &&
-               (ForwardIterator<I> ||
-                (InputIterator<O> && Same<iter_value_t<I>, iter_value_t<O>>) ||
-                IndirectlyCopyableStorable<I, O>)
+    template<input_iterator I, sentinel_for<I> S, weakly_incrementable O,
+             class Proj = identity, indirect_relation<projected<I, Proj>> C = ranges::equal_to>
+      requires indirectly_copyable<I, O> &&
+               (forward_iterator<I> ||
+                (input_iterator<O> && same_to<iter_value_t<I>, iter_value_t<O>>) ||
```
Comment (CaseyCarter) [inline source/algorithms.tex]:
```suggestion
                (input_iterator<O> && same_as<iter_value_t<I>, iter_value_t<O>>) ||
```

Comment (CaseyCarter) [reply source/algorithms.tex]: and on 1667.

File: source/numerics.tex [hunk 4]
```diff
@@ -1947,17 +1947,17 @@
...
 template<class G>
-  concept UniformRandomBitGenerator =
-    Invocable<G&> && UnsignedIntegral<invoke_result_t<G&>> &&
+  concept uniform_random_bit_generator =
+    invocable<G&> && unsigned_integral<invoke_result_t<G&>> &&
     requires {
-      { G::min() } -> Same<invoke_result_t<G&>>;
-      { G::max() } -> Same<invoke_result_t<G&>>;
+      { G::min() } -> same_to<invoke_result_t<G&>>;
+      { G::max() } -> same_to<invoke_result_t<G&>>;
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/numerics.tex]: Doh!  not sure where same_to came from :)

File: source/algorithms.tex [hunk 5]
```diff
@@ -808,16 +808,16 @@
                     BinaryPredicate pred);
 
   namespace ranges {
-    template<InputIterator I1, Sentinel<I1> S1, ForwardIterator I2, Sentinel<I2> S2,
+    template<input_iterator I1, sentinel_for<I1> S1, forward_iterator I2, sentinel_for<I2> S2,
              class Proj1 = identity, class Proj2 = identity,
-             IndirectRelation<projected<I1, Proj1>,
-                              projected<I2, Proj2>> Pred = ranges::equal_to>
+             indirect_relation<projected<I1, Proj1>,
+                                projected<I2, Proj2>> Pred = ranges::equal_to>
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/algorithms.tex]: thanks for catching these!

File: source/algorithms.tex (hunk 2 above)
Comment (burblebee) [reply source/algorithms.tex]: Was trying to create some consistency with how other params were wrapped, but I like your suggestion - fixed.

File: source/algorithms.tex (hunk 3 above)
Comment (burblebee) [reply source/algorithms.tex]: I've done a global change for this, thanks.

File: source/algorithms.tex [hunk 6]
```diff
@@ -3848,8 +3854,8 @@
 \complexity
 No applications of the corresponding predicate and projections if:
 \begin{itemize}
-\item \tcode{S1} and \tcode{I1} model \libconcept{SizedSentinel},
-\item \tcode{S2} and \tcode{I2} model \libconcept{SizedSentinel}, and
+\item \tcode{S1} and \tcode{I1} model \libconcept{sized_sentinel_for},
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/algorithms.tex]: This isn't part of the paper, but I went ahead and made this change in a separate commit.  It's a general problem that needs to be fixed throughout the wording - please open an issue for the general case.  Thanks!

File: source/concepts.tex [hunk 7]
```diff
@@ -269,21 +269,21 @@
...
 
-\indexlibrary{\idxcode{same_to}}%
+\indexlibrary{\idxcode{same_as}}%
 \begin{itemdecl}
 template<class T, class U>
   concept @\placeholdernc{same-impl}@ = is_same_v<T, U>;    // \expos
 
 template<class T, class U>
-  concept same_to = @\placeholdernc{same-impl}@<T, U> && @\placeholdernc{same-impl}@<U, T>;
+  concept same_as = @\placeholdernc{same-impl}@<T, U> && @\placeholdernc{same-impl}@<U, T>;
```
Comment (johelegp) [inline source/concepts.tex]: It seems appropriate to rename _`same-impl`_ to _`same-as-impl`_.

Comment (burblebee) [reply source/concepts.tex]: Combined this with a couple other placeholder name changes mentioned in #3015 as a separate commit.

**Comment by CaseyCarter** [comment]
Thanks again for all the work on this, @burblebee, I really should have prepared this PR myself.

**Comment by burblebee** [comment]
@jwakely you marked this "changes requested" - I addressed all of your concerns afaik.  Might you please have a quick look to make that go away?  I don't like the red :)

**Comment by burblebee** [comment]
@zygoloid this motion is not complete - must be completed after all other motions are merged.

File: source/iterators.tex [hunk 8]
```diff
@@ -1121,13 +1121,13 @@
...
-model \libconcept{SwappableWith}\iref{concept.swappable},
+\libconcept{readable}, and if the reference types of \tcode{E1} and \tcode{E2}
+model \libconcept{swappable_with}\iref{concept.swappable},
 then \tcode{ranges::swap(*E1, *E2)}.
 
 \item Otherwise, if the types \tcode{T1} and \tcode{T2} of \tcode{E1} and
-\tcode{E2} model \tcode{IndirectlyMovableStorable<T1, T2>} and
-\tcode{IndirectlyMovableStorable<T2, T1>}, then
+\tcode{E2} model \tcode{\libconcept{indirectly_movable_storable}<T1, T2>} and
+\tcode{\libconcept{indirectly_movable_storable}<T2, T1>}, then
```
[the remark this answers is not in the document]
Comment (burblebee) [reply source/iterators.tex]: Don't know, but it seemed most existing Latex used \tcode{\libconcept{foo}<T1, T2>}, so I went with that for consistency.

[the remark this answers is not in the document]
Comment (CaseyCarter) [reply source/iterators.tex]: I used `\tcode{\libconcept{C}<T, U>}` in the existing wording for consistency with `\tcode{otherstuff && \libconcept{C}<T, U>}`; `\tcode{otherstuff && }\libconcept{C}\tcode{<T, U>}` feels "weird" to me.

**Comment by zygoloid** [comment]
Rebasing this turned out to be more work than recreating the semi-mechanical renaming part, so I did that and kept all the other changes. This is superseded by https://github.com/cplusplus/draft/pull/3149.

**Pull request closed without merging**